docs/coding-conventions.md

```diff
@@ -320,6 +320,98 @@ coding conventions](#detailed-coding-conventions).
       subdirectories.
 - Document directories and filenames should use dashes rather than underscores.
 
+# Writing Tests
+## Writing functional tests
+### Test Interrupts
+To make sure that any asynchronous operation is canceled when the test is
+terminated, make sure to use golang contexts properly. See more details in 
+[ginkgo documentation](https://onsi.github.io/ginkgo/#spec-timeouts-and-interruptible-nodes)
+
+Avoid using new contexts (e.g. `context.Background()` or `context.TODO()`).
+Instead, prefer using the context injected by ginkgo.
+
+First, add the `ctx` optional parameter of the anonymous functions in `It`, `BeforeEach`,
+`AfterEach` or `DescribeTable`. Then use this parameter for all the contexts in this container.
+
+By adding the `ctx` parameter, we making ginkgo to inject a context.
+
+For example:
+```go
+It("should test something", func(ctx context.Context){
+	...
+	kv, err = virtClient.KubeVirt(kvName).Update(ctx, kv, metav1.UpdateOptions{})
+	Expect(err).ToNot(HaveOccurred())
+	...
+})
+```
+
+When using asynchronous assertion, i.e. `Eventually` or `Consistently`, pass the container's context
+using the `WithContext` method, and add the context parameter to the checked function; e.g.
+```go
+It("should test something", func(ctx context.Context){
+	...
+	
+	Eventually(func(ctx context.Context) error {
+		kv, err = virtClient.KubeVirt(kvName).Update(ctx, kv, metav1.UpdateOptions{})
+		...
+		
+		return err
+	}).WithTimeout(120 * time.Seconds).
+		WithPolling(10 * time.Seconds).
+		WithContext(ctx).
+		Should(Succeed())
+	
+	...
+})
+```
+
+Notice that in case of also passing the `Gomega` parameter, the context parameter will be the second one, i.e.
+```go
+It("should test something", func (ctx context.Context){
+	...
+	
+	Eventually(func(g Gomega, ctx context.Context) {
+		kv, err = virtClient.KubeVirt(kvName).Update(ctx, kv, metav1.UpdateOptions{})
+		g.Expect(err).ToNot(HaveOccurred())
+		...
+	}).WithTimeout(120 * time.Seconds).
+		WithPolling(10 * time.Seconds).
+		WithContext(ctx).
+		Should(Succeed())
+	
+	...
+})
+```
+
+Also notice that ginkgo cancels the context when exiting the container. Avoid using the same
```

Comment:
`Also, notice that gingko cancels contexts when exiting a container, therefore avoid using the same context across multiple containers.`